lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
> in what form do you receive the property?

Do you mean the type of `AST_ObjectKeyVal.key`? Except when they are getter/setter, they can be one of the value types listed in https://github.com/mishoo/UglifyJS2/blob/a0c3836ba0e3baca8d758d9855144ac15ec0ac8d/lib/parse.js#L1359-L1364 - the documentation in `lib/ast.js` wrongly states that `key` converted to a string.